src/home/form/TutorSignUpForm.jsx

```diff
@@ -124,28 +126,45 @@ class SignupTutor extends React.Component {
     }
 
     handleSubmit(event) {
-        alert('A name was submitted: ' + this.state.firstName + ' ' + this.state.lastName);
-        axios.post('/api/registerTutor', this.state)
-            .then(function (response) {
-                console.log(response);
-                if (response.data) {
-                    document.location.href = '/home/login';
-                    console.log("registration successful");
-                } else {
-                    //registration failed
-                    
-                }
-                
-            })
-            .catch(function (error) {
-                console.log(error);
-            });
         event.preventDefault();
+        if (this.state.firstNameValidation === 'input-error' ||
+            this.state.lastNameValidation === 'input-error' ||
+            this.state.usernameValidation === 'input-error' ||
+            this.state.passwordValidation === 'input-error' ||
+            this.state.confirmPasswordValidation === 'input-error' ||
+            this.state.emailValidation === 'input-error' ||
+            this.state.confirmEmailValidation === 'input-error') {
+            this.setState({inputErrorMessage:'error-message'});
+            this.setState({errorMessage:'error-message-hide'});
```

Comment:
Interesting. We might need to change this in the future when we have different types of errors.

Reply:
Yeah another thing I just realized is that I'm not checking for blank inputs in required fields. That whole logic should probably be put into a function.

Reply:
isn't Teju's checks on the front end handling that?